### PR TITLE
fix building greedy_algorithms/ bin 

### PR DESCRIPTION
**Pull request by xgupta**
#### Description of Change

Fix building greedy_algorithm/ bin due name compflict of dijkstra from graph. See issue [#1211](https://github.com/TheAlgorithms/C-Plus-Plus/issues/1211)
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1253"></a>

### CODE REVIEW

**Comment by kvedala** [comment]
Fix one file at a time. Working on the entire folder will be a waste of energy and time. 
Use the provided and configured tool shown in the image below. Using it, you can perform CI checks on individual files before they are merged. All details for how to invoke the static test analyzer are provided in the contribution guidelines: https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#static-code-analyzer
image

**Comment by xgupta** [comment]
> Fix one file at a time.

Ok, I am fixing one file at a time.
 
I am getting an error while running clang-tidy -
/home/runner/work/C-Plus-Plus/C-Plus-Plus/greedy_algorithms/knapsack.cpp:73:13: error: Value stored to 'capacity' is never read [clang-analyzer-deadcode.DeadStores,-warnings-as-errors]
            capacity = 0;

I search on the internet about this error but not find a solution to fix it. Do you have any idea how to solve it?

File: greedy_algorithms/knapsack.cpp [hunk 1]
```diff
@@ -58,21 +61,21 @@ int main() {
...
+            capacity -= static_cast<float>(itemArray[i].weight);
+            std::cout << "\n\t" << itemArray[i].weight << "\t"
+                      << itemArray[i].profit;
         } else {
             maxProfit += profitPerUnit(itemArray[i]) * capacity;
-            cout << "\n\t" << capacity << "\t"
-                 << profitPerUnit(itemArray[i]) * capacity;
+            std::cout << "\n\t" << capacity << "\t"
+                      << profitPerUnit(itemArray[i]) * capacity;
             capacity = 0;
```
Comment (Pardeep009) [inline greedy_algorithms/knapsack.cpp:73]: on removing line 73, will remove error and result will be same.

**Comment by xgupta** [comment]
closing for now and send pull requests for one file at a time.

**Pull request closed without merging**